frontend/packages/metal3-plugin/src/features.ts

```diff
@@ -0,0 +1,19 @@
+import * as _ from 'lodash';
+import { k8sGet, K8sResourceKind } from '@console/internal/module/k8s';
+import { InfrastructureModel } from '@console/internal/models';
+import { getInfrastructurePlatform } from '@console/shared/src/selectors/infrastructure';
+import { setFlag, handleError } from '@console/internal/actions/features';
+import { ActionFeatureFlagDetector } from '@console/plugin-sdk';
+
+export const BAREMETAL = 'BAREMETAL';
+
+export const detectBaremetalPlatform: ActionFeatureFlagDetector = (dispatch) =>
+  k8sGet(InfrastructureModel, 'cluster').then(
+    (infra: K8sResourceKind) =>
+      dispatch(setFlag(BAREMETAL, getInfrastructurePlatform(infra) === 'BareMetal')),
+    (err) => {
+      if (_.get(err, 'response.status') === 404) {
```

Comment:
Can we use `?.` operator here as well?

By doing that, TS compiler can actually see which properties we are accessing. (By using `_.get` we basically throw TS checking away, which is why we should avoid `_.get` if possible.)

Reply:
```ts
err?.response?.status === 404
  ? dispatch(setFlag(BAREMETAL, false))
  : handleError(err, BAREMETAL, dispatch, detectBaremetalPlatform);
```